### ISSUE TITLE
Create GurbaniNowWrapper

### ISSUE THREAD

**Issue by bogas04** [bogas04/SikhJS#72]
It'll be awesome to have a wrapper class with built-in in-memory cache that can also be serialized and initialized. All API calls can then be handled by it in a much efficient manner.

Would be sensible to release as separate module.

**Comment by bogas04** [comment]
``` js
/* https://github.com/rsms/js-lru */

export default class GurbaniNow {
  constructor({ useCache = false, cacheSize = 10, cacheData = { } }) {
    this.constants = {
      API_URL: `https://api.gurbaninow.com`,
      SOURCES: {
        ALL: '',
        SGGS: 'G',
        DG: 'D',
        BGV: 'B',
        BNV: 'N',
        AK: 'A',
        UD: 'U',
      },
      TYPES: {
        FLS: 0,
        FLA: 1,
        WG: 3,
        WE: 4,
      },
    };
    this.cacheEnabled = useCache;

    if (this.cacheEnabled) {
      this.cache = new LRUCache(cacheSize);
      // .. initialize with cacheData 
    }
  }

  search(options) {
    const {
      q, src = this.SOURCES.SGGS, type = this.TYPES.FLS,
      writer, raag, ang, results
    } = options;

    const live = fetch(`${this.constants.API_URL}/search/${q}/...`).then(r => r.json());

    /* update cache anyway */
    live.then(data => this.cacheEnabled && (
      this.cache.put(this.getKey({ route: 'search', ...options }), data)
    ));

    /* assuming cache.get returns promise */
    return (
      this.cacheEnabled
        ? this.cache.get(this.getKey({ route: 'search', ...options })) || live
        : live
    );
  }

  ang({ ang, src = this.SOURCES.SGGS, type = this.TYPES.FLS }) {
    ...
  }

  shabad(id) {
    ...
  }

  hukamnama() {
    ...
  }

  getKey(options) {
    return JSON.stringify(options); // idiotic
  }
}
```

**Comment by bogas04** [comment]
Then a React Component for above

``` jsx
<GurbaniNow q={this.state.q} render={data => <SearchView ...data />} />
<GurbaniNow shabadID={this.state.ID} render={data => <ShabadView ...data />} />
<GurbaniNow hukamnama render={data => <HukamnamaView ...data />} />
<GurbaniNow ang={this.state.ang} source="D" render={data => <AngView ...data />} />
```

**Comment by bogas04** [comment]
React component has been implemented in 637acf3b060c0d6c11a30871ee4b36611bd44a21.
Now we can implement the cache system in it individually.

**Comment by bogas04** [comment]
https://bogas04.github.io/khajana-js/ wraps ~~GurbaniNow~~ Khajana API. Caching can now be done in KhajanaJS itself.
